### PR TITLE
Add doc about how to specify metadata (binaries, services or config).

### PR DESCRIPTION
**Pull request by dholbach**
LP: #1535905

### CODE REVIEW

File: docs/metadata.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      apparmor: app2.apparmor
+      seccomp: app2.seccomp
+```
+
+So in the example, *service1* will declare its start to the relative path
+`bin/app1` and use the standard security template, while *service2* defines
+a different start to `bin/app2`, declares an explicit stop mechanism and uses
+customized **security-policy**, `app2.apparmor` and `app2.seccomp` are
+relative paths to these security artifacts within your project. More about
+security templating can be read about in the [security section] [sec].
```
Comment (kyrofa) [inline docs/metadata.md]: I suggest adding a note that only simple daemons (i.e. non-forking) are supported in 1.x.

File: docs/metadata.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+## Snap config
+
+Snappy config is an entry point to configure your snap, in order to have the
+integration with the config system in place, the **config** keyword needs to
+be used. This is a relative path to the binary that will manage the
+configuration for the snap. If *config* is the part that would handle this
+and building its source creates `bin/config`, the extract would look like
```
Comment (kyrofa) [inline docs/metadata.md]: This section needs a link to "read more about Snappy config" or something.

**Comment by kyrofa** [comment]
Couple thoughts, but overall looks good.

**Comment by sergiusens** [comment]
I'm :+1: on this.

**Pull request merged**